Remove Export Services for Geneva Release

michaelestrin opened the pull request:
Dependencies on:
- https://github.com/edgexfoundry/developer-scripts/pull/186
- https://github.com/edgexfoundry/blackbox-testing/pull/346
- https://github.com/edgexfoundry/go-mod-core-contracts/pull/186

Post-merge of https://github.com/edgexfoundry/go-mod-core-contracts/pull/186, this PR should be updated to
reflect the new version of go-mod-core-contracts.


Fixes: https://github.com/edgexfoundry/edgex-go/issues/2218

Signed-off-by: Michael Estrin <m.estrin@dell.com>

lenny-intel commented on cmd/config-seed/res/properties/edgex-support-rulesengine/application.properties:
```diff
@@ -25,24 +25,6 @@ heart.beat.msg=Support Rules Engine data heart beat
 app.open.msg=This is the Support Rules Engine Microservice.
 # set port (override Spring boot default port 8080 )
 server.port=48075
-#-----------------Export Service Config----------------------------------------
```
This section need to be retained, reworked and renamed to `App Service Configurable Rues config`. See below for required elements.

lenny-intel commented on cmd/config-seed/res/properties/edgex-support-rulesengine;docker/application.properties:
```diff
@@ -26,24 +26,6 @@ heart.beat.msg=Support Rules Engine data heart beat
 app.open.msg=This is the Support Rules Engine Microservice.
 # set port (override Spring boot default port 8080 )
 server.port=48075
-#-----------------Export Service Config----------------------------------------
```
This section need to be retained, reworked and renamed to `App Service Configurable Rues config`. See below for required elements.

lenny-intel commented on cmd/config-seed/res/properties/edgex-support-rulesengine/application.properties:
```diff
@@ -25,24 +25,6 @@ heart.beat.msg=Support Rules Engine data heart beat
-export.client.registration.name=EdgeXRulesEngine
-#use port 5566 when connected to export distro
-#use port 5563 when connected to core data directly
-export.zeromq.port=5566
```
Must be retained

lenny-intel commented on cmd/config-seed/res/properties/edgex-support-rulesengine/application.properties:
```diff
@@ -25,24 +25,6 @@ heart.beat.msg=Support Rules Engine data heart beat
-#use port 5563 when connected to core data directly
-export.zeromq.port=5566
-#export.zeromq.port=5563
-export.zeromq.host=tcp://localhost
```
Must be retained

lenny-intel commented on cmd/config-seed/res/properties/edgex-support-rulesengine/application.properties:
```diff
@@ -25,24 +25,6 @@ heart.beat.msg=Support Rules Engine data heart beat
-#-----------------Export Service Config----------------------------------------
-#Turn on/off registration of rules engine as export distro client.  
-#Set to false to receive messages directly from core data
-export.client=false
```
Should be retained

lenny-intel commented on cmd/config-seed/res/properties/edgex-support-rulesengine;docker/application.properties:
```diff
@@ -26,24 +26,6 @@ heart.beat.msg=Support Rules Engine data heart beat
-#-----------------Export Service Config----------------------------------------
-#Turn on/off registration of rules engine as export distro client.  
-#Set to false to receive messages directly from core data
-export.client=false
```
Should be retained

lenny-intel commented on cmd/config-seed/res/properties/edgex-support-rulesengine;docker/application.properties:
```diff
@@ -26,24 +26,6 @@ heart.beat.msg=Support Rules Engine data heart beat
-export.client.registration.name=EdgeXRulesEngine
-#use port 5566 when connected to export distro
-#use port 5563 when connected to core data directly
-export.zeromq.port=5566
```
Must be retained

lenny-intel commented on cmd/config-seed/res/properties/edgex-support-rulesengine;docker/application.properties:
```diff
@@ -26,24 +26,6 @@ heart.beat.msg=Support Rules Engine data heart beat
-export.zeromq.port=5566
-#export.zeromq.port=5563
-#export.zeromq.host=tcp://localhost
-export.zeromq.host=tcp://edgex-app-service-configurable-rules
```
Must be retained

anonymouse64 commented:
snap changes look ok, need to look more deeply when I am back from traveling, but the build failed due to network nonsense, so restarting the build as necessary should resolve those

lenny-intel commented on cmd/config-seed/res/properties/edgex-support-rulesengine/application.properties:
```diff
@@ -25,24 +25,17 @@ heart.beat.msg=Support Rules Engine data heart beat
-export.client.registration.retry.time=10000
-#how many times to try registration before exiting
-export.client.registration.retry.attempts=100
+#export.zeromq.host=tcp://edgex-app-service-configurable-rules
```
This one needs to be `localhost`

michaelestrin replied:
I think it is; see the line immediately before this one.  I preserved the commented out lines that provide the alternate configuration settings (e.g. in this case, for Docker).

michaelestrin commented:
Rebased.

michaelestrin commented:
Rebased.

AnthonyMBonafide commented:
Is this affected by these changes?

https://github.com/edgexfoundry/edgex-go/blob/1e7bb157cf0c15e1831a4e3b03a3216743e1f80d/internal/core/data/event.go#L134

It doesn't look to me like we need to do anything, but I am not 100% sure.

lenny-intel commented:
> Is this affected by these changes?
> 
> https://github.com/edgexfoundry/edgex-go/blob/1e7bb157cf0c15e1831a4e3b03a3216743e1f80d/internal/core/data/event.go#L134
> 
> It doesn't look to me like we need to do anything, but I am not 100% sure.

Just need to reword the comment to something like "Push event to message bus for App Services to consume"

michaelestrin commented:
Rebased after https://github.com/edgexfoundry/edgex-go/pull/2223 was merged.  Addressed Anthony's comments.

michaelestrin commented:
recheck

anonymouse64 commented:
@tonyespy will be reviewing this so removing myself for review

jim-wang-intel commented on cmd/config-seed/res/properties/edgex-support-rulesengine/application.properties:
```diff
@@ -25,24 +25,17 @@ heart.beat.msg=Support Rules Engine data heart beat
-export.client.registration.retry.time=10000
-#how many times to try registration before exiting
-export.client.registration.retry.attempts=100
+#export.zeromq.host=tcp://edgex-app-service-configurable-rules
```
should remove this line as the configuration already in the line above and it is kind of duplicate info for it already being in docker's version as well.

michaelestrin replied:
Won't fix.  This was left in intentionally and follows the established pattern of keeping docker and non-docker options in both variations of configuration files (with the non-current variation being commented out).

jim-wang-intel replied:
ok, understanding the established pattern, but it is misleading info to left here IMHO. As configuration in the local, majority of times, it is `localhost` always

jim-wang-intel commented on internal/seed/config/populate.go:
```diff
@@ -160,9 +160,15 @@ func ImportSecurityConfiguration() error {
-	var names = [9]string{clients.CoreMetaDataServiceKey, clients.CoreCommandServiceKey, clients.CoreDataServiceKey,
-		clients.ExportDistroServiceKey, clients.ExportClientServiceKey, clients.SupportLoggingServiceKey,
-		clients.SupportSchedulerServiceKey, clients.SupportNotificationsServiceKey, clients.SystemManagementAgentServiceKey}
+	var names = [9]string{
```
should be `7` now instead of `9`

michaelestrin replied:
Great catch!

jim-wang-intel commented on internal/seed/config/populate.go:
```diff
@@ -160,9 +160,15 @@ func ImportSecurityConfiguration() error {
 
 // As services are converted to utilize V2 types, add them to this list and remove from the one above.
 func listDirectories() [9]string {
```
should be `7` now instead of `9`

michaelestrin replied:
Great catch!

michaelestrin commented:
Rebased.  Addressed fixed array size in internal/seed/config/populate.go that was not updated when elements were removed (@jim-wang-intel's review comments).

michaelestrin commented:
@tsconn23 @tonyespy 

Intent is to merge this by EOD Tuesday (tomorrow).  Please try to review and comment/approve before then.
